### PR TITLE
Spark 3.3: Remove use of deprecated SparkFilesScan

### PR DESCRIPTION
**Pull request by szehon-ho**
Follow up of : https://github.com/apache/iceberg/pull/6924

### CODE REVIEW

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ class SparkFilesScan extends SparkScan {
   @Override
   protected List<CombinedScanTask> taskGroups() {
     if (tasks == null) {
-      FileScanTaskSetManager taskSetManager = FileScanTaskSetManager.get();
```
Comment (aokolnychyi) [inline spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: Can we remove this class and `SparkFilesScanBuilder` completely?

Comment (aokolnychyi) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: Shall we also drop `FileScanTaskSetManager`?

Comment (aokolnychyi) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: Oh, wait, 1.2 is not out yet.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java [hunk 2]
```diff
@@ -260,11 +260,6 @@ public MetadataColumn[] metadataColumns() {
 
   @Override
   public ScanBuilder newScanBuilder(CaseInsensitiveStringMap options) {
-    if (options.containsKey(SparkReadOptions.FILE_SCAN_TASK_SET_ID)) {
```
Comment (aokolnychyi) [inline spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java:263]: We should either keep this or wait until 1.2 is out and then remove all classes and deprecated vars.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java (hunk 1 above)
Comment (aokolnychyi) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: If we plan to merge before 1.2, it is not safe to change `SparkFilesScan`.

Comment (szehon-ho) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: I'm ok to wait until 1.2 is merged

Comment (nastra) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: I agree that this PR should only be merged once 1.2.0 is officially out. Also would be great to delete `FileScanTaskSetManager` as part of this PR

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java [hunk 3]
```diff
@@ -97,7 +97,7 @@ public void testBinPackRewrite() throws NoSuchTableException, IOException {
       FileRewriteCoordinator rewriteCoordinator = FileRewriteCoordinator.get();
       Set<DataFile> rewrittenFiles =
           taskSetManager.fetchTasks(table, fileSetID).stream()
-              .map(FileScanTask::file)
+              .map(t -> ((FileScanTask) t).file())
```
Comment (nastra) [inline spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java:100]:
```suggestion
              .map(t -> t.asFileScanTask().file())
```

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java [hunk 4]
```diff
@@ -167,7 +167,7 @@ public void testSortRewrite() throws NoSuchTableException, IOException {
...
```
Comment (nastra) [inline spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java:170]:
```suggestion
              .map(t -> t.asFileScanTask().file())
```

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java [hunk 5]
```diff
@@ -243,7 +243,7 @@ public void testCommitMultipleRewrites() throws NoSuchTableException, IOExceptio
     Set<DataFile> rewrittenFiles =
         fileSetIDs.stream()
             .flatMap(fileSetID -> taskSetManager.fetchTasks(table, fileSetID).stream())
-            .map(FileScanTask::file)
+            .map(t -> ((FileScanTask) t).file())
```
Comment (nastra) [inline spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java:246]: same as above

**Review by nastra — APPROVED** [review]
overall LGTM, but we can only merge the PR **after** 1.2.0 is officially out. Also I think we can delete `FileScanTaskSetManager` as part of this PR.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java (hunk 2 above)
Comment (szehon-ho) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java]: Restored this usage.

File: spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java (hunk 1 above)
Comment (szehon-ho) [reply spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkFilesScan.java:57]: Removed in latest commit

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java (hunk 3 above)
Comment (szehon-ho) [reply spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/TestFileRewriteCoordinator.java]: Good idea, thanks

**Comment by aokolnychyi** [comment]
Let me take another look.

**Review by aokolnychyi — APPROVED** [review]

**Comment by aokolnychyi** [comment]
Thanks, @szehon-ho! Thanks for reviewing, @nastra!

**Pull request merged**